core/src/main/java/io/stanwood/framework/arch/core/rx/ResourceStatusTransformer.kt

```diff
@@ -44,25 +44,27 @@ object ResourceStatusTransformer {
     fun <T> fromFlowable(): FlowableTransformer<Resource<T>, ResourceStatus> =
         FlowableResourceTransformer()
 
-    private fun <T> toStatus(source: Resource<T>): ResourceStatus =
-        source.data?.let { ResourceStatus.Success } ?: when (source) {
-            is Resource.Failed -> ResourceStatus.Error(source.msg, source.cause)
-            is Resource.Loading -> ResourceStatus.Loading
-            else -> ResourceStatus.Success
-        }
-
     private class FlowableResourceTransformer<T> : FlowableTransformer<Resource<T>, ResourceStatus> {
         override fun apply(upstream: Flowable<Resource<T>>): Publisher<ResourceStatus> =
-            upstream.map { toStatus(it) }
+            upstream.map { it.toResourceStatus() }
     }
 
     private class ObservableResourceTransformer<T> : ObservableTransformer<Resource<T>, ResourceStatus> {
         override fun apply(upstream: Observable<Resource<T>>): ObservableSource<ResourceStatus> =
-            upstream.map { toStatus(it) }
+            upstream.map { it.toResourceStatus() }
     }
 
     private class SingleResourceTransformer<T> : SingleTransformer<Resource<T>, ResourceStatus> {
         override fun apply(upstream: Single<Resource<T>>): SingleSource<ResourceStatus> =
-            upstream.map { toStatus(it) }
+            upstream.map { it.toResourceStatus() }
     }
-}
\ No newline at end of file
+}
+
+fun <T> Resource<T>.toResourceStatus() =
+    data
+        ?.let { ResourceStatus.Success }
```

Comment:
Not sure I understand this (I know, it has been like this before, but I only now recognized that I in fact don't understand this :D ) - why does a non-null data always result in a success? Given the current resource definition the other states also *may* have a non-null data.

Reply:
You are right , i need to rework this in next version.